### PR TITLE
StudentList implementation v1.9

### PR DESCRIPTION
**Pull request by Valelaaa**
- fixed junit tests

### CODE REVIEW

File: src/test/java/com/endava/internship/collections/StudentListTest.java [hunk 1]
```diff
@@ -56,399 +56,495 @@ public void Init() {
...
+        studentList.add(STUDENT1);
+        studentList.add(STUDENT2);
+        listIterator = studentList.listIterator();
         listIterator.next();
-        assertTrue(listIterator.hasPrevious());
+        assertTrue(listIterator.hasPrevious(),"List doesn't have previous element");
     }
 
     @Test
     void previousShouldReturnsCurrentElement() {
```
Comment (NataliaIovva) [inline src/test/java/com/endava/internship/collections/StudentListTest.java:412]: rename to shouldReturnFirstElementOfList

File: src/test/java/com/endava/internship/collections/StudentListTest.java [hunk 2]
```diff
@@ -56,399 +56,495 @@ public void Init() {
...
+                "ToArray returns wrong array");
     }
 
-    @ParameterizedTest
-    @NullSource
-    void toArrayThrowsNullPointerExceptionIfArrayIsNull(final Object[] object) {
-        assertThrows(NullPointerException.class, () -> students.toArray(object));
+    @Test
+    void shouldThrowNullPointerExceptionWhenToArrayGetsNull() {
+        assertThrows(NullPointerException.class, () -> students.toArray((Object[]) null),
```
Comment (NataliaIovva) [inline src/test/java/com/endava/internship/collections/StudentListTest.java:200]: please remove redundant cast `(Object[])`

File: src/test/java/com/endava/internship/collections/StudentListTest.java [hunk 3]
```diff
@@ -56,399 +56,495 @@ public void Init() {
...
+        studentList.add(STUDENT1);
+        studentList.add(STUDENT2);
+        iterator = studentList.iterator();
 
-    @Test
-    void shouldNextShouldReturnCurrentElementOfList() {
-        assertEquals(STUDENT0, iterator.next());
+        assertTrue(iterator.hasNext(),
+                "List doesn't have next element");
+        for (int i = 0; iterator.hasNext() && i < DEFAULT_SIZE;i++ )
```
Comment (NataliaIovva) [inline src/test/java/com/endava/internship/collections/StudentListTest.java:167]: try to avoid cycles in tests

Comment (Valelaaa) [reply src/test/java/com/endava/internship/collections/StudentListTest.java:167]: understood!

File: src/test/java/com/endava/internship/collections/StudentListTest.java (hunk 2 above)
Comment (Valelaaa) [reply src/test/java/com/endava/internship/collections/StudentListTest.java:200]: Done

File: src/test/java/com/endava/internship/collections/StudentListTest.java (hunk 1 above)
Comment (Valelaaa) [reply src/test/java/com/endava/internship/collections/StudentListTest.java:412]: Done

**Pull request closed without merging**